Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndoHelpers.cpp

```diff
@@ -17,12 +17,16 @@ namespace AzToolsFramework
         namespace PrefabUndoHelpers
         {
             void UpdatePrefabInstance(
-                const Instance& instance, AZStd::string_view undoMessage, const PrefabDom& instanceDomBeforeUpdate,
+                Instance& instance, AZStd::string_view undoMessage, const PrefabDom& instanceDomBeforeUpdate,
                 UndoSystem::URSequencePoint* undoBatch)
             {
                 PrefabDom instanceDomAfterUpdate;
                 PrefabDomUtils::StoreInstanceInPrefabDom(instance, instanceDomAfterUpdate);
 
+                // Preemptively updates the cached dom to prevent unnecessary reloading in InstanceSerializer::Reload().
```

Comment:
nit: remove function name from the comment as implementation may be moved to a different function later and the comment would be incorrect.